opensrp-app/src/main/java/org/smartregister/repository/P2PReceiverTransferDao.java

```diff
@@ -49,7 +49,7 @@ public long receiveJson(@NonNull DataType dataType, @NonNull JSONArray jsonArray
         EventClientRepository foreignEventClientRepository = CoreLibrary.getInstance().context().getForeignEventClientRepository();
 
         int eventsMaxRowId = eventClientRepository.getMaxRowId(eventClientRepository.getEventTable());
-        int foreignEventsMaxRowId = foreignEventClientRepository.getMaxRowId(foreignEventClientRepository.getEventTable());
+        int foreignEventsMaxRowId = (getP2PClassifier() == null) ? 0 : foreignEventClientRepository.getMaxRowId(foreignEventClientRepository.getEventTable());
```

Comment:
could you use `CoreLibrary.getInstance().context().hasForeignEvents()` for consistency